rcl/src/rcl/client.c

```diff
@@ -288,16 +288,20 @@ rcl_take_response(
   return RCL_RET_OK;
 }
 
-bool rcl_client_is_valid(const rcl_client_t * client)
+bool rcl_client_is_valid(const rcl_client_t * client, rcl_allocator_t * error_msg_allocator)
 {
   const rcl_client_options_t * options;
-  RCL_CHECK_ARGUMENT_FOR_NULL(
-    client, false, rcl_get_default_allocator());
+  const rcl_allocator_t alloc =
+    error_msg_allocator ? *error_msg_allocator : rcl_get_default_allocator();
+  RCL_CHECK_ALLOCATOR_WITH_MSG(&alloc, "error msg allocator is invalid", return false);
```

Comment:
The error message seems to be misleading in case `error_msg_allocator` was a `nullptr`.

Same below.

Reply:
Please make a suggestion, or feel free to push an updated error message directly to this branch if it's faster

Reply:
`allocator is invalid` is the simplest if we don't want to distinguish the two cases.